### PR TITLE
Update the vsphere sample config

### PR DESCRIPTION
**Pull request by danielnelson**
This change syncs the sample config in the program with the readme, fixes a fix minor copy/paste typos, and updates the style to more closely follow the guidelines on the [wiki](https://github.com/influxdata/telegraf/wiki/SampleConfig).

related #5639

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
@prydin Can you review?

File: etc/telegraf.conf [hunk 1]
```diff
@@ -5070,14 +5070,21 @@
...
 #   username = "user@corp.local"
+#
+#   ## vCenter Password
 #   password = "secret"
 #
-#   ## VMs
-#   ## Typical VM metrics (if omitted or empty, all metrics are collected)
+#   ## Inventory paths for the virtual machines to gather metrics on.  If empty,
+#   ## all VMs are included.
+#   ##   ex: vm_include = ["/*/vm/**"]
```
Comment (prydin) [inline etc/telegraf.conf]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: etc/telegraf.conf [hunk 2]
```diff
@@ -5114,11 +5121,17 @@
...
+#
+#   ## If true, gather virtual machine instance metrics.
+#   # vm_instances = true
 #
-#   ## Hosts
-#   ## Typical host metrics (if omitted or empty, all metrics are collected)
+#   ## Inventory path of the hosts to gather metrics on.  If empty, all hosts
+#   ## are included.
+#   ##   ex: host_include = [ "/*/host/**"]
+#   # host_include = []
```
Comment (prydin) [inline etc/telegraf.conf]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: etc/telegraf.conf [hunk 3]
```diff
@@ -5167,49 +5180,73 @@
...
-#   ## Plugin Settings
-#   ## separator character to use for measurement and field names (default: "_")
+#   host_metric_exclude = []
+#
+#   ## If true, gather host instance metrics.
+#   # host_instances = true
+#
+#   ## Inventory paths for the clusters to gather metrics on.  If empty,
+#   ## all clusters are included.
+#   ##   ex: cluster_include = ["/*/cluster/**"]
```
Comment (prydin) [inline etc/telegraf.conf]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: etc/telegraf.conf [hunk 4]
```diff
@@ -5167,49 +5180,73 @@
...
+#   ## Cluster metrics to gather.
+#   cluster_metric_include = []
+#   cluster_metric_exclude = []
+#
+#   ## If true, gather cluster instance metrics.
+#   # cluster_instances = false
+#
+#   ## Inventory paths for the datastores to gather metrics on.  If empty,
+#   ## all datastores are included.
+#   ##   ex: datastore_include = ["/*/datastore/**"]
```
Comment (prydin) [inline etc/telegraf.conf]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: etc/telegraf.conf [hunk 5]
```diff
@@ -5167,49 +5180,73 @@
...
+#   ## Datastore metrics to gather.
+#   datastore_metric_include = []
+#   datastore_metric_exclude = []
+#
+#   ## If true, gather datastore instance metrics.
+#   # datastore_instances = false
+#
+#   ## Inventory paths for the datacenter to gather metrics on.  If empty,
+#   ## all datacenter are included.
+#   ##   ex: datacenter_include = ["/*/datacenter/**"]
```
Comment (prydin) [inline etc/telegraf.conf]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/README.md [hunk 6]
```diff
@@ -17,17 +17,22 @@ vm_metric_exclude = [ "*" ]
...
+  ## vCenter Password
   password = "secret"
 
-  ## VMs
-  ## Typical VM metrics (if omitted or empty, all metrics are collected)
-  # vm_include = [ "/*/vm/**"] # Inventory path to VMs to collect (by default all are collected)
+  ## Inventory paths for the virtual machines to gather metrics on.  If empty,
+  ## all VMs are included.
+  ##   ex: vm_include = ["/*/vm/**"]
+  # vm_include = []
```
Comment (prydin) [inline plugins/inputs/vsphere/README.md]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/README.md [hunk 7]
```diff
@@ -64,12 +69,17 @@ vm_metric_exclude = [ "*" ]
...
 
-  ## Hosts
-  ## Typical host metrics (if omitted or empty, all metrics are collected)
-  # host_include = [ "/*/host/**"] # Inventory path to hosts to collect (by default all are collected)
+  ## If true, gather virtual machine instance metrics.
+  # vm_instances = true
+
+  ## Inventory path of the hosts to gather metrics on.  If empty, all hosts
+  ## are included.
+  ##   ex: host_include = [ "/*/host/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/README.md]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/README.md [hunk 8]
```diff
@@ -118,52 +128,73 @@ vm_metric_exclude = [ "*" ]
...
-  ## Plugin Settings
-  ## separator character to use for measurement and field names (default: "_")
+  host_metric_exclude = []
+
+  ## If true, gather host instance metrics.
+  # host_instances = true
+
+  ## Inventory paths for the clusters to gather metrics on.  If empty,
+  ## all clusters are included.
+  ##   ex: cluster_include = ["/*/cluster/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/README.md]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/README.md [hunk 9]
```diff
@@ -118,52 +128,73 @@ vm_metric_exclude = [ "*" ]
...
+  ## Cluster metrics to gather.
+  cluster_metric_include = []
+  cluster_metric_exclude = []
+
+  ## If true, gather cluster instance metrics.
+  # cluster_instances = false
+
+  ## Inventory paths for the datastores to gather metrics on.  If empty,
+  ## all datastores are included.
+  ##   ex: datastore_include = ["/*/datastore/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/README.md]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/README.md [hunk 10]
```diff
@@ -118,52 +128,73 @@ vm_metric_exclude = [ "*" ]
...
+  ## Datastore metrics to gather.
+  datastore_metric_include = []
+  datastore_metric_exclude = []
+
+  ## If true, gather datastore instance metrics.
+  # datastore_instances = false
+
+  ## Inventory paths for the datacenter to gather metrics on.  If empty,
+  ## all datacenter are included.
+  ##   ex: datacenter_include = ["/*/datacenter/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/README.md]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/vsphere.go [hunk 11]
```diff
@@ -155,49 +168,73 @@ var sampleConfig = `
...
-  ## Plugin Settings
-  ## separator character to use for measurement and field names (default: "_")
+  host_metric_exclude = []
+
+  ## If true, gather host instance metrics.
+  # host_instances = true
+
+  ## Inventory paths for the clusters to gather metrics on.  If empty,
+  ## all clusters are included.
+  ##   ex: cluster_include = ["/*/cluster/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/vsphere.go]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/vsphere.go [hunk 12]
```diff
@@ -155,49 +168,73 @@ var sampleConfig = `
...
+  ## Cluster metrics to gather.
+  cluster_metric_include = []
+  cluster_metric_exclude = []
+
+  ## If true, gather cluster instance metrics.
+  # cluster_instances = false
+
+  ## Inventory paths for the datastores to gather metrics on.  If empty,
+  ## all datastores are included.
+  ##   ex: datastore_include = ["/*/datastore/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/vsphere.go]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

File: plugins/inputs/vsphere/vsphere.go [hunk 13]
```diff
@@ -155,49 +168,73 @@ var sampleConfig = `
...
+  ## Datastore metrics to gather.
+  datastore_metric_include = []
+  datastore_metric_exclude = []
+
+  ## If true, gather datastore instance metrics.
+  # datastore_instances = false
+
+  ## Inventory paths for the datacenter to gather metrics on.  If empty,
+  ## all datacenter are included.
+  ##   ex: datacenter_include = ["/*/datacenter/**"]
```
Comment (prydin) [inline plugins/inputs/vsphere/vsphere.go]: Make this path "/**" instead. Gets rid of that permission issue someone ran into a while ago. Requires a change to the default struct initializer in vsphere.go. See #5613

**Comment by danielnelson** [comment]
I'll update the struct initialization code.  Is `/**` treated the same as `datacenter_include = []`?  If so, I'll use `[]` as the default values.   Since these `ex:` values are just examples, maybe we can come up with some better example values that match real uses?

**Comment by prydin** [comment]
Hmm... Actually ```[ ]``` means "nothing is included". That may not be right, since it's not consistent with the metric includes. So the examples should read ```[ "/**" ]``` at least until we've decided what the correct handling of ```[ ]``` should be.

**Comment by danielnelson** [comment]
It doesn't seem to work this way, `[]` appears to function as include all and that seems to be backed up by the current documentation.  This is in line with other Telegraf plugins.

**Comment by danielnelson** [comment]
@prydin You are right about it being "nothing included" since we don't go into `Find(` at all if the list is empty.  Are you okay with me changing `[]` to collect everything?  This will be more expected behavior based on our other plugins.

https://github.com/influxdata/telegraf/blob/495a5e9f99bc4acb6bc45c00fdcf0726acb98928/plugins/inputs/vsphere/finder.go#L39-L42

**Comment by prydin** [comment]
@danielnelson Yes, you may make that change.

**Comment by danielnelson** [comment]
This is rather out of date, will reopen pr if I can find time.

**Pull request closed without merging**